### PR TITLE
boards: arm: Add steval_fcu001v1 board support

### PR DESCRIPTION
**Pull request by Navin-Sankar**
BSP for steval_fcu001v1 resubmitted as per PR #18746 

Signed-off-by: NavinSankar Velliangiri <navin@linumiz.com>

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -83,6 +83,7 @@
 /boards/arm/v2m_beetle/                   @fvincenzo
 /boards/arm/olimexino_stm32/              @ydamigos
 /boards/arm/sensortile_box/               @avisconti
+/boards/arm/steval_fcu001v1/		  @Navin-Sankar
```
Comment (erwango) [inline CODEOWNERS]: Can you check indentation?

File: boards/arm/steval_fcu001v1/Kconfig.board [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Kconfig - STM32 Flight Controller Unit board configuration
+#
+# Copyright (c) 2019 Linumiz
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BOARD_STEVAL_FCU001V1
+	bool "STM32 Flight Controller Unit"
+	depends on SOC_STM32F401XE
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/Kconfig.board]: SOC_STM32F401XC

File: boards/arm/steval_fcu001v1/steval_fcu001v1.dts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	};
+};
+
+
+&i2c2 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+&spi1 {
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/steval_fcu001v1.dts]: Haven't seen spi configuration elsewhere, please remove.

File: boards/arm/steval_fcu001v1/steval_fcu001v1_defconfig [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_ARM=y
+CONFIG_BOARD_STEVAL_FCU001V1=y
+CONFIG_SOC_SERIES_STM32F4X=y
+CONFIG_SOC_STM32F401XE=y
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/steval_fcu001v1_defconfig]: CONFIG_SOC_STM32F401XC

File: boards/arm/steval_fcu001v1/support/openocd.cfg [hunk 5]
```diff
@@ -0,0 +1,18 @@
+source [find interface/stlink.cfg]:q
```
Comment (erwango) [inline boards/arm/steval_fcu001v1/support/openocd.cfg]: `:q` looks like a left over.

**Review by erwango — CHANGES_REQUESTED** [review]
Clean work!
Though, you miss Kconfig definition for the SoC variant (STM32F401XC), please add it as a separate commit (you can add the .dtsi addition as well in this commit).
Thanks!

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**